doc/src/manual/performance-tips.md

```diff
@@ -1394,91 +1366,109 @@ a = rand(Float32,1000) * 1.f-9
-      goto 28
-      19:
-      unless (y::UNION{FLOAT64,INT64} isa Float64)::ANY goto 23
-      #temp#@_6::Core.MethodInstance = MethodInstance for *(::Float64, ::Float64)
-      goto 28
+      # meta: location bool.jl & 40
+      Core.SSAValue(15) = (Base.and_int)(Core.SSAValue(14), false)::Bool
+      # meta: pop location
+      # meta: location bool.jl | 41
+      Core.SSAValue(16) = (Base.or_int)(Core.SSAValue(12), Core.SSAValue(15))::Bool
+      # meta: pop locations (2)
+      unless Core.SSAValue(16) goto 21
+      #temp#@_5::UNION{FLOAT64, INT64} = 0
+      goto 23
+      21:
+      #temp#@_5::UNION{FLOAT64, INT64} = x::Float64
       23:
-      goto 25
-      25:
-      #temp#@_7::Float64 = (y::UNION{FLOAT64,INT64} * x::Float64)::Float64
-      goto 30
-      28:
-      #temp#@_7::Float64 = $(Expr(:invoke, :(#temp#@_6), :(Main.*), :(y), :(x)))
-      30:
-      return $(Expr(:invoke, MethodInstance for sin(::Float64), :(Main.sin), :((Base.add_float)(#temp#@_7,(Base.sitofp)(Float64,1)::Float64)::Float64)))
+      # meta: pop location
+      y::UNION{FLOAT64, INT64} = #temp#@_5::UNION{FLOAT64, INT64}
+      #= line 3 =#
+      Core.SSAValue(17) = (y::UNION{FLOAT64, INT64} isa Float64)::Bool
+      unless Core.SSAValue(17) goto 31
+      #temp#@_6::Float64 = $(Expr(:invoke, MethodInstance for *(::Float64, ::Float64), :(Main.:*), :(y), :(x)))::Float64
+      goto 40
+      31:
+      Core.SSAValue(18) = (y::UNION{FLOAT64, INT64} isa Int64)::Bool
+      unless Core.SSAValue(18) goto 36
+      #temp#@_6::Float64 = $(Expr(:invoke, MethodInstance for *(::Int64, ::Float64), :(Main.:*), :(y), :(x)))::Float64
+      goto 40
+      36:
+      goto 38
+      38:
+      (Base.error)("fatal error in type inference (type bound)")::UNION{}
+      40:
+      Core.SSAValue(0) = #temp#@_6::Float64
+      # meta: location promotion.jl + 312
+      # meta: location float.jl + 391
+      Core.SSAValue(41) = (Base.add_float)(Core.SSAValue(0), 1.0)::Float64
+      # meta: pop locations (2)
+      Core.SSAValue(2) = $(Expr(:invoke, MethodInstance for sin(::Float64), :(Main.sin), Core.SSAValue(41)))::Float64
+      return Core.SSAValue(2)
   end::Float64
 ```
 
 Interpreting the output of [`@code_warntype`](@ref), like that of its cousins [`@code_lowered`](@ref),
 [`@code_typed`](@ref), [`@code_llvm`](@ref), and [`@code_native`](@ref), takes a little practice.
 Your code is being presented in form that has been partially digested on its way to generating
-compiled machine code.  Most of the expressions are annotated by a type, indicated by the `::T`
+compiled machine code. Most of the expressions are annotated by a type, indicated by the `::T`
 (where `T` might be [`Float64`](@ref), for example). The most important characteristic of [`@code_warntype`](@ref)
 is that non-concrete types are displayed in red; in the above example, such output is shown in
 all-caps.
 
 The top part of the output summarizes the type information for the different variables internal
```

Comment:
The text here described output from ´@code_warntype` in a way earlier julia version than what we actually showed here. So someone likely updated the output but not the text below.